File: src/main/java/work/lclpnet/ap2/game/anvil_fall/AnvilFallMiniGame.java
package work.lclpnet.ap2.game.anvil_fall;

import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.util.Identifier;
import work.lclpnet.ap2.api.game.GameType;
import work.lclpnet.ap2.api.game.MiniGame;
import work.lclpnet.ap2.api.game.MiniGameHandle;
import work.lclpnet.ap2.api.game.MiniGameInstance;
import work.lclpnet.ap2.base.ApConstants;
import work.lclpnet.ap2.base.ArcadeParty;

public class AnvilFallMiniGame implements MiniGame {

    @Override
    public Identifier getId() {
        return ArcadeParty.identifier("anvil_fall");
    }

    @Override
    public GameType getType() {
        return GameType.FFA;
    }

    @Override
    public String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public ItemStack getIcon() {
        return new ItemStack(Items.ANVIL);
    }

    @Override
    public boolean canBeFinale() {
        return true;
    }

    @Override
    public boolean canBePlayed() {
        return true;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new AnvilFallInstance(gameHandle);
    }
}

File: src/minigames/anvil_fall/src/main/kotlin/work/lclpnet/ap2/game/anvil_fall/AnvilFallMiniGame.kt
package work.lclpnet.ap2.game.anvil_fall

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.util.MapLevelGameFactory

class AnvilFallMiniGame : MiniGame {
    override val id = ApConstants.identifier("anvil_fall")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.ANVIL)
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = true
    override fun createFactory() = MapLevelGameFactory(::AnvilFallInstance)
}
